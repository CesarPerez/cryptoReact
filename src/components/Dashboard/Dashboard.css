.container-dashboard {
  padding: 3em;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-content: center;
  justify-content: space-evenly;
  align-items: center;
  background: #f3f6f9;
  height: 82vh;
}

.desc-1 {
  text-align: initial;
}
