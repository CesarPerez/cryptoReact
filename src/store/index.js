// import { composeWithDevTools } from "redux-devtools-extension";
import counterSlice from './reducers/counter';
import { configureStore, combineReducers } from '@reduxjs/toolkit';
import authSlice from './reducers/auth';
import notifSlice from './reducers/notif';

const allReducers = combineReducers({
  counter: counterSlice.reducer,
  auth: authSlice.reducer,
  notif: notifSlice.reducer,
});

export const store = configureStore({
  reducer: allReducers,
});

export default store;
