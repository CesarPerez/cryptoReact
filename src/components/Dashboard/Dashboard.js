import { useCallback, useEffect, useState } from 'react';
import Button from '@mui/material/Button';
import FloodIcon from '@mui/icons-material/Flood';
import './Dashboard.css';
import { useDispatch, useSelector } from 'react-redux';
import { counterActions } from '../../store/reducers/counter';

function Dashboard() {
  const dispatch = useDispatch();

  const clickAdd = useCallback(() => {
    dispatch(counterActions.increment());
  }, [dispatch]);

  const counter = useSelector(state => state.counter.counter);

  return (
    <>
      <div className="container-dashboard">
        <section>
          <div className="desc-1">
            <h3>Add Cryptos to your wishlish</h3>
            <p>
              You can note some of the cryptos you want to track in the future, there are new ones every day, dont miss
              the opportunity!
            </p>
          </div>
        </section>
        <div>
          <Button onClick={clickAdd} variant="contained" endIcon={<FloodIcon />}>
            Add randomly
          </Button>
          <p id="msg" data-testid="msg">
            {counter}
          </p>
        </div>
      </div>
    </>
  );
}

export default Dashboard;
