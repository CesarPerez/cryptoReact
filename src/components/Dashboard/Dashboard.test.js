import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store';
import Dashboard from './Dashboard';
import renderer from 'react-test-renderer'; // ES6

test('Should show a button and a counter', () => {
  render(
    <Provider store={store}>
      <Dashboard />
    </Provider>,
  );
  const butt = screen.getByText('Add randomly');
  const txt = screen.getByTestId('msg');
  expect(butt).toBeInTheDocument();
  expect(txt).toBeInTheDocument();
});

test('Should match snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Dashboard />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
