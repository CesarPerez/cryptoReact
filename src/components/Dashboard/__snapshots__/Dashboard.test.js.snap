// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div
  className="container-dashboard"
>
  <section>
    <div
      className="desc-1"
    >
      <h3>
        Add Cryptos to your wishlish
      </h3>
      <p>
        You can note some of the cryptos you want to track in the future, there are new ones every day, dont miss the opportunity!
      </p>
    </div>
  </section>
  <div>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Add randomly
      <span
        className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="FloodIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18.67 19c-1.95 0-2.09 1-3.33 1-1.19 0-1.42-1-3.33-1-1.95 0-2.1 1-3.34 1-1.24 0-1.38-1-3.33-1-1.95 0-2.1 1-3.34 1v2c1.95 0 2.11-1 3.34-1 1.24 0 1.38 1 3.33 1 1.95 0 2.1-1 3.34-1 1.22 0 1.4 1 3.33 1 1.93 0 2.1-1 3.33-1 1.22 0 1.4 1 3.33 1v-2c-1.24 0-1.38-1-3.33-1zm-9.99-1.5c1.95 0 2.09-1 3.33-1 1.19 0 1.42 1 3.33 1 1.95 0 2.09-1 3.33-1 1.19 0 1.4.98 3.31 1v-2c-.63 0-1-.28-1.48-.55l-2.02-7.53 2.09.85.74-1.86L9.78 2 2 11.61l1.57 1.23 1.39-1.78.93 3.48c-.18-.02-.35-.05-.56-.05-1.95 0-2.09 1-3.33 1v2c1.9 0 2.17-1 3.35-1 1.19.01 1.42 1.01 3.33 1.01zm5.36-7.32 1.42 5.31c-1.34.09-1.47-.99-3.47-.99-.36 0-.65.04-.91.1l-.91-3.39 3.87-1.03z"
          />
        </svg>
      </span>
    </button>
    <p
      data-testid="msg"
      id="msg"
    >
      1
    </p>
  </div>
</div>
`;
